src/WebJobs.Extensions.DurableTask/FunctionFailedException.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// The exception that is thrown when a sub-orchestrator or activity function fails
+    /// with an error.
+    /// </summary>
+    public class FunctionFailedException : Exception
```

Comment:
>public [](start = 4, length = 6)

internal?

Reply:
This is intentionally public. I want developers to be able to explicitly catch this exception in a try/catch block (otherwise they have to catch all exceptions, which isn't always desirable). I've updated the `TryCatchLoop` test orchestration to explicitly catch this exception type to make sure it's being surfaced correctly. The trace validation code should be looking for it as well.